internal/backend/runtime/omni/controllers/omni/cluster_machine_config.go

```diff
@@ -450,8 +456,8 @@ func buildInstallImage(clusterMachineTalosVersion *omni.ClusterMachineTalosVersi
 	schematicID := schematicConfig.Id
 
 	secureBootStatus := machineStatus.TypedSpec().Value.SecureBootStatus
-	if secureBootStatus == nil {
-		return "", xerrors.NewTaggedf[qtransform.SkipReconcileTag]("secure boot status for machine %q is not yet set", machineStatus.Metadata().ID())
+	if secureBootStatus == nil { // should never happen - must have been handled before entering this function
+		return "", fmt.Errorf("machine %q has no secure boot status set", machineStatus.Metadata().ID())
```

Comment:
do we need this check then?

Reply:
we don't really need it, but isn't it still better than accidentally switching from secureboot to non-secureboot if we ever use this function from a 3rd place in the future?

Reply:
This function is internal for this controller.
I don't think we should ever call it from the other places.

Reply:
We already call it from two places - while generating config from `ClusterMachineConfigController` and while upgrading a node from `ClusterMachineConfigStatusController`.

Reply:
ahh, that's `buildInstallImage`. I got confused.

Reply:
ignore me then :)) I thought it's part of reconcileMachineConfig